### ISSUE TITLE
Windows 2012r2 Server : "Exception: Could not load file or assembly"

### ISSUE THREAD

**Issue by bushy555** [rainforestnet/CrystalReportsNinja#27]
Hi,

Have been running Ninja on Win 2003 for the past few years.
* Attempting to convert over from Win 2003 to Windows Server 2012R2.
* Have attempted both your 32 and 64 bit versions on Win 2012 R2.
* Win Server 2012R2 is completley up to date with patches.
* Dot net 4.5 does not install since it is already part of the operating system.
* CRRuntime_64bit_13_0_20.msi is installed.

Have tried everything , including a few dumb 'out-there' things to attempt to get ti running.
Unfortunately everything I try simply throws the error message : 

Exception: Could not load file or assembly 'CrystalDecisions.CrystalReports.Engine, Version=13.0.3500.0, Culture=neutral, PublicKeyToken=692fbea5521e1304' or on
e of its dependencies. The system cannot find the file specified.
Inner Exception:

Any pointers?

**Comment by MarkFirman** [comment]
Exactly the same issue here. Not even deployed onto a server yet; still on the development machine. I have installed all relevant patches ect. Error "Could not load file or assembly 'CrystalDecisions.CrystalReports.Engine" The system cannot find the file specified. I have definitely installed the correct runtime packages ect.

**Comment by bushy555** [comment]
...I managed to get it working surprisingly after many many fiddlings of just simply "doing stuff" after many many hours. I can't narrow it down exactly what it was that I did to make it work.

1. Found the older rainforest.net software; installed, ran, and still failed.
2. Installed our old copy of a full version of Crystal Reports v9.
3. Attempted to force install Dot.Net v2.5, 3.0, v4.5, and latest v4.7.1 on this dopey 64 bit Windows 2012r2 server. Most failed with "Already installed on this o/s", but one or two installed successfully.
Installed Java and Adobe Flash player just for the hell of it. I was grasping at straws by now.
4. Rebooted. Still had errors.
5. Re-worked the parameters on the crystalreportsninja.exe file.
    Getting weird error messgages left right and centre when trying to connect to our SQL server and Informix database.
Kept trying. Finally got a working connection to our Informix database.
Used the same format of parameters that worked on the informix database on our SQL database, but still had the same COMM errors, crystal errors, and Assembly errors. Frustrating !!
6. Our SQL server has around 8 different SQL databases under various instances. Everything I threw at it just didnt work.  But the Informix database was still all fine and dandy.

Physical server hostname : SQL1
The database I was after :  SQL1\TRACER_SQL

Then I tried something totally dumb , using parameters: 
-S TRACER_SQL -D TRACER_SQL
...without the instance name and without the physical server hostname. And the damn thing worked.
It doesn't make any sense, but it worked. Appears as though it is also case sensitive.

(....using 32 bit SQL ODBC drivers under 'System' on a 64 bit server.)

....best of luck.
